### PR TITLE
Fix/collection point inheritance

### PR DESCRIPTION
**Pull request by gastonabella**
### Brief Description
Single Table Inheritance implemented for CollectionPoints -> Container/Event
### Related card or ticket
https://github.com/Pis-moove-it/reciclando-backend/issues/110
### Additional Details
* STI implemented for CollectionPoints
* Event model created
* Organization added to CollectionPoints
* Tests and factories fixed because of these changes

### CODE REVIEW

File: spec/controllers/collections_controller_spec.rb [hunk 1]
```diff
@@ -21,26 +21,26 @@ def create_collection_call(route_id, collection_point_id, pocket_attributes)
 
       context 'when creating valid collections' do
         it 'does return success' do
-          create_collection_call(auth_route.id, collection_point.id, serial_numbers)
+          create_collection_call(auth_route.id, container.id, serial_numbers)
```
Comment (GonzaNunezCano) [inline spec/controllers/collections_controller_spec.rb]: Maybe you could avoid repeating the create_collection_call in this context, adding before(:each) { create_collection_call(auth_route.id, container.id, serial_numbers) }

File: app/models/container.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
-class Container < ApplicationRecord
+class Container < CollectionPoint
   enum status: %i[Ok Damaged Removed]
-  validates :status, presence: true
+  validates :status, :active, presence: true
+
+  before_validation(on: :create) do
+    self.active = true
```
Comment (fbalarini) [inline app/models/container.rb:6]: this can be set as the default value

File: db/seeds.rb [hunk 3]
```diff
@@ -48,12 +48,6 @@
   end
 end
 
-unless CollectionPoint.count.positive?
```
Comment (GonzaNunezCano) [inline db/seeds.rb:51]: Please remove the line 12 to permit execute the rails db:seed command

**Review by fbalarini — APPROVED** [review]

File: app/controllers/collections_controller.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 class CollectionsController < AuthenticateController
   def create
     return render_error(1, 'Missing pockets') if collection_params[:pockets_attributes].blank?
+    return render_error(1, 'Invalid collection_point_id') if container.blank?
```
Comment (fbalarini) [inline app/controllers/collections_controller.rb]: `unless container` should work

**Review by fbalarini — APPROVED** [review]

**Pull request merged**